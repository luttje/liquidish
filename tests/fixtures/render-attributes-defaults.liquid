{% meta {
    "isChildOnly": true,
    "defaults": {
        "attributes": "",
        "slot": "Button",
        "complex": {
            "human": {
                "name": "Anne",
                "age": 25,
                "isOnline": false
            }
        },
        "overrideDefault": false
    }
} %}
<button class="px-4 py-2 self-start text-sm whitespace-nowrap font-medium text-white bg-[#c70f19] border-bottom-[#980b13] hover:bg-[#df111c] hover:border-bottom-[#800a10] rounded-md shadow uppercase"
        {{ attributes }}>
    {{ slot }}{% if slot == 'Button' %}!{% endif %}

    {% if complex.human %}
        <span>Welcome</span>
    {% endif %}

    {% if overrideDefault %}
        Override
    {%endif%}

    {% if complex.human.isOnline %}
        <span>Online</span>
    {% endif %}

    {% if complex.human.name == 'Anne' %}
        <span>{{ complex.human.name }}</span>
    {% endif %}
    {% if complex.human.age < '25' %}
        <span>{{ complex.human.age }}</span>
    {% elsif complex.human.age == '25' %}
        <span>Cool</span>
    {% else %}
        <span>Not cool</span>
    {% endif %}

    {% unless complex.human.name %}
        <span>Wont Show</span>
    {% endunless %}
</button>
