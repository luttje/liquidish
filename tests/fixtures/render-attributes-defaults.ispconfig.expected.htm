
<button class="px-4 py-2 self-start text-sm whitespace-nowrap font-medium text-white bg-[#c70f19] border-bottom-[#980b13] hover:bg-[#df111c] hover:border-bottom-[#800a10] rounded-md shadow uppercase"
        >
    Button{tmpl_if name="___" op="!=" value="show-it"}!{/tmpl_if}

    {tmpl_if name="___" op="!=" value="show-it"}
        <span>Welcome</span>
    {/tmpl_if}

    {tmpl_if name="___" op="!=" value="show-it"}
        Override
    {/tmpl_if}

    {tmpl_if name="___" op="==" value="dont-show-it"}
        <span>Online</span>
    {/tmpl_if}

    {tmpl_if name="___" op="!=" value="show-it"}
        <span>Anne</span>
    {/tmpl_if}
    {tmpl_if name="___" op="==" value="dont-show-it"}
        <span>25</span>
    {tmpl_elseif name="___" op="!=" value="show-it"}
        <span>Cool</span>
    {tmpl_else}
        <span>Not cool</span>
    {/tmpl_if}

    {tmpl_unless name="___" }
        <span>Wont Show</span>
    {/tmpl_unless}
</button>
